pixeltable/catalog/path.py

```diff
@@ -55,5 +56,16 @@ def is_ancestor(self, other: Path, is_parent: bool = False) -> bool:
         is_prefix = self.components == other.components[: self.len]
         return is_prefix and (self.len == (other.len - 1) or not is_parent)
 
+    def ancestors(self) -> Iterator[Path]:
+        """
+        Return all ancestors of this path in top-down order including root.
+        If this path is for the root directory, which has no parent, then None is returned.
+        """
+        if self.is_root:
```

Comment:
Why is this necessary as a special case? I'd expect that if `self.is_root`, then `len(self.components) == 0`, in which case it's handled correctly by the regular logic. (Or is there a component with value `''`?)

Reply:
@property
 def is_root(self) -> bool:
        return self.components[0] == ''
from above I assumed that there is possibility that len is 1 with component[0]=''